GUI/src/gui_qt.py

```diff
@@ -0,0 +1,458 @@
+import threading
+import shutil
+from datetime import datetime
+import yaml
+from PySide6.QtWidgets import (
+    QApplication, QMainWindow, QPushButton, QComboBox, QLabel, QTextEdit, QFileDialog, QVBoxLayout, QHBoxLayout, QToolBar, QDialog, QWidget, QDialogButtonBox
+)
+from PySide6.QtCore import Qt
+from multiprocessing import Process, Queue
+
+class FileSelectDialog(QDialog):
+    def __init__(self, title, files, parent=None):
+        super().__init__(parent)
+        self.setWindowTitle(title)
+        self.setModal(True)
+        layout = QVBoxLayout()
+        
+        self.combo = QComboBox()
+        self.combo.addItems(files)
+        layout.addWidget(self.combo)
+        
+        button_box = QDialogButtonBox(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
+        button_box.accepted.connect(self.accept)
+        button_box.rejected.connect(self.reject)
+        layout.addWidget(button_box)
+        
+        self.setLayout(layout)
+        
+    def get_selected(self):
+        return self.combo.currentText() if self.result() == QDialog.Accepted else None
+
+class SimulatorGUI(QMainWindow):
+    def __init__(self):
+        super().__init__()
+        self.setWindowTitle("CDA Simulator GUI")
+        self.setMinimumSize(600, 400)
+        
+        self.repo_folder = None
+        self.selected_folder = None
+        self.config_folder = None
+        self.start_script = None
+        self.stop_script = None
+        self.docker_compose_file = None
+        self.build_script = None
+        self.log_queue = Queue()  # For multiprocessing log messages
+        
+        # Check and create /tmp/.env if it doesn't exist
+        env_file = "/tmp/.env"
+        if not os.path.exists(env_file):
+            try:
+                with open(env_file, "w") as f:
+                    f.write("# Dummy .env file for stop_simulation\n")
+                print(f"[GUI Log] Created {env_file}")
+            except Exception as e:
+                print(f"[GUI Log] Failed to create {env_file}: {e}")
+        
+        # Main widget and layout
+        self.central_widget = QWidget()
+        self.setCentralWidget(self.central_widget)
+        self.main_layout = QVBoxLayout()
+        self.central_widget.setLayout(self.main_layout)
+        
+        # Toolbar
+        self.toolbar = QToolBar("Main Toolbar")
+        self.addToolBar(self.toolbar)
+        
+        # Browse button
+        self.browse_button = QPushButton("Browse Repo Folder")
+        self.browse_button.clicked.connect(self.browse_folder)
+        self.toolbar.addWidget(self.browse_button)
+        
+        # Open CARMA Logs button
+        self.carma_logs_button = QPushButton("Open CARMA Logs")
+        self.carma_logs_button.clicked.connect(lambda: self.open_logs("/opt/carma/logs"))
+        self.toolbar.addWidget(self.carma_logs_button)
+        
+        # Open CARMA-Simulation Logs button
+        self.carma_sim_logs_button = QPushButton("Open CARMA-Simulation Logs")
```

Comment:
This is the CDASim mosaic logs folder right? I would name it as such. Maybe "CDASim Logs" CARMA-Simulation is a name we've moved away from so it would be good to avoid it to minimize confusion.